gnucash/CMakeLists.txt

```diff
@@ -44,6 +45,46 @@ TARGET_LINK_LIBRARIES (gnucash
    ${GUILE_LDFLAGS} ${GLIB2_LDFLAGS} ${GTK3_LDFLAGS} ${GTK_MAC_LDFLAGS}
 )
 
+# Get glib executable for generating the gresource file
+EXECUTE_PROCESS(
+   COMMAND
+       ${PKG_CONFIG_EXECUTABLE} gio-2.0 --variable glib_compile_resources
+   OUTPUT_VARIABLE
+       GLIB_COMPILE_RESOURCES_EXECUTABLE
+   OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Get the dependencies of the gresource
+EXECUTE_PROCESS(
+  OUTPUT_VARIABLE
+      gr_files
+  WORKING_DIRECTORY
+     ${CMAKE_CURRENT_SOURCE_DIR}
+  COMMAND ${GLIB_COMPILE_RESOURCES_EXECUTABLE}
+     --generate-dependencies
+     ${CMAKE_CURRENT_SOURCE_DIR}/gnucash-gresources.xml
+)
+
+STRING (REPLACE "\n" ";" gresource_files ${gr_files})
+
+SET (DEPENDS "")
+FOREACH (cur_file ${gresource_files})
+    LIST (APPEND gnucash_GRESOURCE "${cur_file}")
+ENDFOREACH ()
```

Comment:
I don't think this is necessary. Any string with semicolons in it is considered a list by cmake. So you can remove this part work with gresource_files directly everywhere below instead of gnucash_RESOURCE.